docs/src/keps/kep-001.rst

```diff
@@ -0,0 +1,172 @@
+
+
+
+What is a KEP?
+--------------
+
+KEP stands for Klio Enhancement Proposal. A KEP is a design document that: provides information to the Klio community; describes a new feature for the Klio ecosystem; or describes a new process.
+Similar to `standard RFCs`_), a KEP should provide a concise technical specification and rationale for a proposed feature.
+Inspiration was taken from `Backstage's ADRs`_, `Python's PEPs`_, `Jupyter's JEPs`_, `NumPy's NEPs`_, and `OpenJDK's JEPs`_.
+
+We intend KEPs to be the primary mechanism for proposing major new features, for collecting community input on an issue, and for documenting the design decisions that have gone into the Klio ecosystem. The KEP author is responsible for building consensus within the community and documenting dissenting opinions.
+
+Because the KEPs are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal [1]_.
+
+
+Types
+-----
+
+There are three kinds of KEPs:
+
+1. A **Standards Track** KEP describes a new feature or implementation for Klio.
+
+2. An **Informational** KEP describes a Klio design issue, or provides general guidelines or information to the Python community, but does not propose a new feature. Informational KEPs do not necessarily represent a Klio community consensus or recommendation, so users and implementers are free to ignore Informational KEPs or follow their advice.
+
+3. A **Process** KEP describes a process surrounding Klio, or proposes a change to (or an event in) a process. Process KEPs are like Standards Track KEPs but apply to areas other than the Klio language itself. They may propose an implementation, but not to Klio's codebase. Examples include procedures, guidelines, changes to the decision-making process, and changes to the tools or environment used in Klio development. Any meta-KEP is also considered a Process KEP.
+
+
+Workflow
+--------
+
+The KEP process begins with a new idea for Klio.
+It is highly recommended that a single KEP contain a single key proposal or new idea.
+Small enhancements or patches often don't need a KEP and can be injected into the Klio development workflow with a pull request to the `Klio repo`_.
+The more focused the KEP, the more successful it tends to be. If in doubt, split your KEP into several well-focused ones.
+
+Each KEP must have a champion: someone who writes the KEP using the style and format described below, shepherds the discussions in the appropriate forums, and attempts to build consensus around the idea.
+The KEP champion (a.k.a. Author) should first attempt to ascertain whether the idea is suitable for a KEP.
+This may include posting to the #klio channel in the `Spotify FOSS Slack Organization`_.
+
+The proposal should be submitted as a draft KEP via a GitHub `pull request`_ to the ``docs/src/keps`` directory with the name ``kep-<n>.rst`` where ``<n>`` is an appropriately assigned three-digit number (e.g., ``kep-000.rst``).
```

Comment:
Who assigns the number?  Just up to the author to pick the next one?

If a PEP is withdrawn is its number recycled?  Same question for rejected but for that I assume it's not.

Reply:
Good q; I'll update the doc to reflect this, but basically: always increment by 1 (no real reason other than to have an explicit standard; I don't think python PEPs are strictly incremental; numpy NEPs seem to be). And no number recycling (it could be confusing, I'm guessing).